editoast/src/views/timetable/train_schedule.rs

```diff
@@ -590,10 +608,359 @@ async fn occupancy_blocks(
+    train_schedule: &models::TrainSchedule,
+    path_projection: &PathProjection,
+    operational_point_track_offsets: &[TrackOffset],
+    report_train: &ReportTrain,
+) -> Vec<(String, TrackOccupancy)> {
+    // Iterate over the track offsets linked to the operational point
+    // until we find its corresponding positions on the pathfinding route.
+    // Then, retrieve the associated times (via interpolation) and positions on the path.
+    operational_point_track_offsets
+        .iter()
+        .filter_map(|track_offset| {
+            path_projection.get_position(track_offset).map(|position| {
+                let index = find_index_upper(&report_train.positions, position);
+                let time = if index == 0 {
+                    report_train.times[0]
+                } else {
+                    interpolate(
+                        report_train.positions[index - 1],
+                        report_train.positions[index],
+                        report_train.times[index - 1],
+                        report_train.times[index],
+                        position,
+                    )
+                };
+                (
+                    track_offset.track.to_string(),
+                    TrackOccupancy {
+                        time_begin: train_schedule.start_time + Duration::milliseconds(time as i64),
+                        duration: PositiveDuration::default(),
+                        train_schedule_id: train_schedule.id,
+                    },
+                )
+            })
+        })
+        .collect()
+}
+
+fn find_track_occupancy_for_operational_point(
+    operational_point_id: &str,
+    operational_point_track_offsets: &[TrackOffset],
+    path_item_cache: &PathItemCache,
+    simulation: &Response,
+    pathfinding: &PathfindingResult,
+    train_schedule: &models::TrainSchedule,
+) -> Vec<(String, TrackOccupancy)> {
+    // Get the track ranges from the pathfinding
+
+    let mut track_occupancies = vec![];
+    let (track_ranges, path_item_positions) = match pathfinding {
+        PathfindingResult::Success(pathfinding_result_success) => {
+            assert_eq!(
+                pathfinding_result_success.path_item_positions.len(),
+                train_schedule.path.len()
+            );
+            (
+                &pathfinding_result_success.track_section_ranges,
+                &pathfinding_result_success.path_item_positions,
+            )
+        }
+        PathfindingResult::Failure(_) => {
+            // The timetable can have a pathfinding failure for a train schedule
+            // because the train schedule is not valid.
+            // We skip it because we don't want to compute the track occupancy for an invalid train schedule.
+            // It's not a problem because we are just looking for the track occupancy of the operational point for the valid trains.
+            tracing::info!(train_schedule.id, "pathfinding failed");
+            return track_occupancies;
```

Comment:
I would rather directly return `vec![]` here to be more direct about how this error case is handled.